test/jdk/java/lang/ProcessBuilder/Basic.java

```diff
@@ -598,7 +599,15 @@ private static void testEncoding(String encoding, String tested) {
         try {
             // If round trip conversion works, should be able to set env vars
             // correctly in child.
-            if (new String(tested.getBytes()).equals(tested)) {
+            String nativeEncoding = System.getProperty("native.encoding");
+            String fileEncoding = System.getProperty("file.encoding");
+            Charset cs;
+            if (nativeEncoding != null && !nativeEncoding.equals(fileEncoding)) {
```

Comment:
What's the reason for comparing `file.encoding`? Does `Charset.forName(nativeEncoding, Charset.defaultCharset())` suffice?

Reply:
Remove file.encoding related code.